Automattic/simplenote-ios
# Project: In App Notifications - View Implementation and Styling

### Fix
This PR begins the process of implementing in app notifications to Simplenote iOS.  It deals with the first two to do items on  issue #1177 Implement Notice View and enable dark mode styling.

Note that this current setup does not animate the appearance of the notices and is only intended to display them in the correct location and styled correctly.

light mode:
<img src="https://cdn-std.droplr.net/files/acc_593859/TJCocs" />

dark mode:
<img src="https://cdn-std.droplr.net/files/acc_593859/6y9XWQ" />

### Test
There is currently no implementation to display notes, so to test, first go to this [bin](https://mc.a8c.com/pb/28971/) and copy the code there.  Paste it into SPNoteViewController+Extensions.swift starting at line 37 replacing configureNavigationBarItems( ) and adding setup/display functions for the notices

1. once setup with the testing code, open up Simplenote and go to any note.
2. tap on the action button to see a notice without an action, long press on the notification and check the log for a response then wait for it to disappear
3. tap on the create note button to see a notice with an action appear, long press on the notification and check the log for a response, then wait for it to disappear
4. tap on the create note again, and then tap on the "retry" button you should see in the log a message appear after tapping the button
5. Set device to dark mode and repeat steps 2-4 again
6. Test device rotation to make sure that the notice stay centered and just above bottom of the screen
7. This PR also covers accessibility, so make sure to test that the notices appear correctly when dynamic text is enabled

Note that these do not currently account for keyboards, so if the keyboard is present the notification will not be visible. 

### Review
> Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
 These changes do not require release notes.
